projects/dotcms-field-elements/src/components/dot-checkbox/dot-checkbox.tsx

```diff
@@ -1,58 +1,139 @@
+                'dot-valid': this.isValid(),
+                'dot-invalid': !this.isValid(),
+                'dot-pristine': this._dotPristine,
+                'dot-dirty': !this._dotPristine,
+                'dot-touched': this._dotTouched,
+                'dot-untouched': !this._dotTouched
+            }
+        };
+    }
+
+    /**
+     * Reset properties of the field, clear value and emit events.
+     *
+     * @memberof DotSelectComponent
+     */
+    @Method()
+    reset(): void {
+        this._dotPristine = true;
+        this._dotTouched = false;
+        this._valid = true;
+        this.value = '';
+        this.emitValueChange();
+        this.emitStatusChange();
     }
 
     render() {
         return (
             <Fragment>
-                <label htmlFor={this._label}>{this.label}</label>
+                <label htmlFor={this.name}>{this.label}</label>
                 {this._options.map((item: DotOption) => {
-                    this._values = { ...this._values, [item.value]: this.value === item.value ? true : false };
+                    const trimmedValue = item.value.trim();
                     return (
-                        <div class='dot-checkbox__container'>
+                        <div class={this.getClassName()}>
                             <input
                                 type='checkbox'
-                                name={item.value}
-                                checked={this.value === item.value ? true : null}
+                                disabled={this.shouldBeDisabled()}
+                                id={this.value}
+                                checked={this.value.indexOf(trimmedValue) >= 0 ? true : null}
                                 onInput={(event: Event) => this.setValue(event)}
-                                value={item.value}
+                                value={trimmedValue}
                             />
-                            <label htmlFor={item.value}>{item.label}</label>
+                            <label htmlFor={trimmedValue}>{item.label}</label>
                         </div>
                     );
                 })}
-                {this.hint ? <span class='dot-textfield__hint'>{this.hint}</span> : ''}
+                {this.hint ? <span class='dot-field__hint'>{this.hint}</span> : ''}
+                {!this.isValid() ? <span class='dot-field__error-message'>{this.requiredmessage}</span> : ''}
             </Fragment>
         );
     }
+
+    // Todo: find how to set proper TYPE in TS
+    private setValue(event): void {
+        this._dotPristine = false;
+        this._dotTouched = true;
+        this.value = this.getValueFromCheckInputs(event.target.value.trim(), event.target.checked);
+        this.emitValueChange();
+        this.emitStatusChange();
+    }
+
+    private getValueFromCheckInputs(value: string, checked: boolean): string {
+        const valueArray = this.value.trim().length > 0 ? this.value.split(',') : [];
```

Comment:
why don't we keep 'value' as a String array?

Reply:
the `BE` accepts a string with values separated with commas